client/utils/guidance.py

```diff
@@ -0,0 +1,210 @@
+        for each misunderstanding.
+        """    
+
+        #Creates a new folder inside tests that stores the number of misU per assignment
+        if os.path.isfile(self.current_working_dir + COUNT_FILE_PATH):
+            with open(self.current_working_dir + COUNT_FILE_PATH,'r') as f:
+                jsonDic = json.load(f)
+                answerDict = jsonDic["answerDict"]
+                countData = jsonDic["countData"] 
+        else:
+            countData = {}
+            answerDict = {}
+
+        # Checks to see if the question is already stored and whether the student's answer is there
+        # It does not update misU count if same answer was seen before
+        if shorten_unique_id in answerDict and wrongAnswer in answerDict[shorten_unique_id]:
+            return countData
+        if not shorten_unique_id in answerDict:
+            answerDict[shorten_unique_id] = []
+
+        # Updates misU count
+        for x in lst_misU: 
+            if x in countData:
+                countData[x] += 1
+            else:
+                countData[x] = 1
+
+        # Stores the updated count back into the same file and overrides it
+        newjsonDic = {}
+        answerDict[shorten_unique_id].append(wrongAnswer)
+        newjsonDic["countData"] = countData
+        newjsonDic["answerDict"] = answerDict
+        with open(self.current_working_dir + COUNT_FILE_PATH,"w") as f:
+            json.dump(newjsonDic,f)
+        # Returns a dictionary containing how many times each misU has been seen.
+        return countData
+
+    def set_tg(self,access_token,guidance_flag):
+        """
+        Try to grab the treatment group number for the student. If there is no treatment 
+        group number available, we request it from the server.
+        """
+        if guidance_flag:
+            with open(self.current_working_dir + LOCAL_TG,"w") as f:
+                f.write(str(GUIDANCE_FLAG_TG_NUMBER))
+            self.tg_id = GUIDANCE_FLAG_TG_NUMBER
+            return
+        # Checks to see the student currently has a treatment group number. If not, calls helper function in auth.py
+        if not os.path.isfile(self.current_working_dir + LOCAL_TG):
+            cur_email = auth.get_student_email(access_token)
+            if not cur_email:
+                print ("--Not quite. Try again! --")
+                # return ({}, -1)
+                return DEFAULT_RETURN_VALUE # Does it matter to have the empty string at the end?
+            try:
+                data = json.loads(urlopen(TGSERVER + cur_email + TG_SERVER_ENDING,timeout =1).read().decode("utf-8"))
+            except IOError as e:
+                data = {"tg":-1}
+            fd = open(self.current_working_dir + LOCAL_TG,"w")
+            fd.write(str(data["tg"]))
+            fd.close()
+
+        tg_file = open(self.current_working_dir + LOCAL_TG, 'r')
+        self.tg_id = int(tg_file.read())
+
+util = Guidance
```

Comment:
You should remove this variable. The AttributeError is from [your code in unlock.py](https://github.com/Cal-CS-61A-Staff/ok-client/pull/165/files#diff-c66760f0a6aa193db7eca384468b2da0R40), which should read `guidance.Guidance` instead of `guidance.util`